packages/experimental/kai/src/containers/SpaceList/SpaceList.tsx

```diff
@@ -86,43 +91,31 @@ const SpaceItem = withReactor(({ space, selected, children, onAction }: SpaceIte
   );
 });
 
+export type SpaceListAction = {
+  spaceKey: PublicKey;
+  modifier?: boolean;
+};
+
 export type SpaceListProps = {
   spaces: Space[];
   selected?: PublicKey;
-  onSelect: (spaceKey: PublicKey) => void;
-  onShare: (spaceKey: PublicKey) => void;
+  onAction?: (intent: Intent<SpaceListAction>) => void;
 };
 
-// TODO(burdon): Use vertical mosaic.
-export const SpaceList = ({ spaces, selected, onSelect, onShare }: SpaceListProps) => {
+export const SpaceList = ({ spaces, selected, onAction }: SpaceListProps) => {
   const { frame } = useAppRouter();
   const Tile = frame?.runtime.Tile;
 
-  // TODO(burdon): Factor pattern?
-  const handleAction = (spaceKey: PublicKey, action: SpaceItemAction) => {
-    switch (action) {
-      case SpaceItemAction.SELECT: {
-        onSelect(spaceKey);
-        break;
-      }
-
-      case SpaceItemAction.SHARE: {
-        onShare(spaceKey);
-        break;
-      }
-    }
-  };
-
   return (
     <div className='flex flex-col flex-1 overflow-hidden m-2'>
       {spaces.map((space) => (
         <SpaceItem
           key={space.key.toHex()}
           space={space}
           selected={selected && space.key.equals(selected)}
-          onAction={(action) => handleAction(space.key, action)}
+          onAction={(intent) => onAction?.(intent)}
         >
-          {Tile && <Tile />}
+          <Suspense>{Tile && <Tile />}</Suspense>
```

Comment:
Ditto here, though it seems like the best fallback wouldn’t be a spinning loader but a shimmering text skeleton. We should add that to r-c.

Reply:
ok -- you just resolved my issue above :).